lambda/src/main/scala/pricemigrationengine/handlers/NotificationHandler.scala

```diff
@@ -79,16 +79,30 @@ object NotificationHandler extends CohortHandler {
       item: CohortItem,
       subscription: ZuoraSubscription
   ): ZIO[CohortTable with Zuora, Failure, Unit] = {
+    // Known keys for this mapping can be found in the BillingPeriod object
```

Comment:
not very important, but a more robust approach is to use `BillingPeriod.fromString` and have an exhaustive pattern match instead of a Map

Reply:
Actually you are right, let me do that....

Reply:
@tomrf1 
That one did the trick :)
https://github.com/guardian/price-migration-engine/pull/1099/commits/6e16895fb17d7de51b220b798cb42f3b6842ba32

Reply:
ah, I was imagining a match on `BillingPeriod` -
```
def billingPeriodToInt(period: BillingPeriod): Int =
  period match {
    case Monthly => 1
    case Quarterly => 3
    case SemiAnnual => 6
    case Annual => 12
  }
```
So that you can call:
`BillingPeriod.billingPeriodToInt(BillingPeriod.fromString(billingPeriod))`